packages/core/src/client.ts

```diff
@@ -3427,12 +3439,20 @@ export class MedplumClient extends TypedEventTarget<MedplumClientEventMap> {
 
   /**
    * Refreshes the access token using the refresh token if available.
+   * @param gracePeriod - Optional grace period in milliseconds. If not specified, uses the client configured grace period (default 5 minutes).
    * @returns Promise to refresh the access token.
    */
-  refreshIfExpired(): Promise<void> {
+  refreshIfExpired(gracePeriod?: number): Promise<void> {
+    if (gracePeriod === undefined) {
+      gracePeriod = this.refreshGracePeriod;
+    }
     // If (1) not already refreshing, (2) we have an access token, and (3) the access token is expired,
     // then start a refresh.
-    if (!this.refreshPromise && this.accessTokenExpires !== undefined && this.accessTokenExpires < Date.now()) {
+    if (
+      !this.refreshPromise &&
+      this.accessTokenExpires !== undefined &&
+      this.accessTokenExpires < Date.now() - gracePeriod
```

Comment:
This might be
```suggestion
      this.accessTokenExpires - gracePeriod < Date.now()
```
this?